components/compositing/constellation.rs

```diff
@@ -990,11 +990,12 @@ impl<LTF: LayoutThreadFactory, STF: ScriptThreadFactory> Constellation<LTF, STF>
             // then reuse the script thread in creating the new pipeline
             let source_url = &source_pipeline.url;
 
+            // FIXME(#10968): this should probably match the origin check in
+            //                HTMLIFrameElement::contentDocument.
             let same_script = source_url.host() == new_url.host() &&
```

Comment:
That's not correct either; I don't think it's useful to replace one broken check by another.